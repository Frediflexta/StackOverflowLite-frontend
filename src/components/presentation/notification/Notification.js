import React from 'react';
import propTypes from 'prop-types';
// import '../styles/Notification.scss';

const Notification = (props) => {
  const {
    message,
    display,
    status
  } = props;

  const style = {
    display
  };

  let color = '';

  if (status === 'success') {
    color = 'is-success';
  } else if (status === 'error') {
    color = 'is-danger';
  } else {
    color = 'is-info';
  }

  return (
    <div style={style}
      className={`notification ${color}`}
    >
      {message}
    </div>
  );
};

Notification.propTypes = {
  status: propTypes.string,
  message: propTypes.string,
  display: propTypes.string,
};

Notification.defaultProps = {
  status: '',
  message: '',
  display: 'none',
};

export default Notification;
