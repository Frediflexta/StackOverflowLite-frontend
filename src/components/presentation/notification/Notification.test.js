import React from 'react';
import { shallow } from 'enzyme';
import Notification from './Notification';

describe('<Notification /> shallow rendering tests', () => {
  const wrapper = shallow(
    <Notification
      status="success"
      message="This rendered successfully"
      display="block"
    />
  );

  it('should render with props successfully', () => {
    expect(wrapper.find('div').hasClass('is-success')).toBe(true);
    expect(wrapper.find('div').text()).toBe('This rendered successfully');
  });

  it('should render with is-danger class based on the status prop', () => {
    wrapper.setProps({
      status: 'error'
    });
    expect(wrapper.find('div').hasClass('is-danger')).toBe(true);
    expect(wrapper.find('div').hasClass('is-success')).toBe(false);
  });

  it('should render with is-info class when no status prop', () => {
    wrapper.setProps({
      status: null
    });
    expect(wrapper.find('div').hasClass('is-info')).toBe(true);
    expect(wrapper.find('div').hasClass('is-danger')).toBe(false);
    expect(wrapper.find('div').hasClass('is-success')).toBe(false);
  });

  it('matches the snapshot', () => {
    const tree = shallow(<Notification />);
    expect(tree).toMatchSnapshot();
  });
});
