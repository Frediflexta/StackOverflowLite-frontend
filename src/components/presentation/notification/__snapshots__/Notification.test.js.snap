// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> shallow rendering tests matches the snapshot 1`] = `
<div
  className="notification is-info"
  style={
    Object {
      "display": "none",
    }
  }
/>
`;
